### PR TITLE
Add PUP list

### PR DESCRIPTION
**Pull request by iam-py-test**
## Summary
This PR adds the PUP list I requested in https://github.com/blocklistproject/Lists/discussions/449, and adds some domain from my blocklists to start it off.

## Checklist

- [X] I have verified that I have not modified any files inside the `alt-version` folder (automated code will automatically update those files)

- [X] I have verified that I have not modified any files inside the `dnsmasq-version` folder (automated code will automatically update those files)

### CODE REVIEW

**Comment by iam-py-test** [comment]
@blocklistproject there is a lint error, but it does not seem to be a result of my changes

File: pups.txt [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+0.0.0.0 shop.winzip.com
+0.0.0.0 activate.winzip.com
+0.0.0.0 subscribe.winzip.com
+0.0.0.0 install.winzip.com
+0.0.0.0 secure.winzip.com
+0.0.0.0 response.winzip.com
+0.0.0.0 download-origin.winzip.com
+0.0.0.0 donwload.winzip.com
+0.0.0.0 www6.winzip.com
+0.0.0.0 openvpn.winzip.com
```
Comment (spirillen) [inline pups.txt:112]: `winzip.com`?? as I recall it winzip is a rather widely used packing program for shitdows!

Comment (iam-py-test) [reply pups.txt:112]: @spirillen let me check.
I took these from my lists, so there may be a false positive

**Comment by spirillen** [comment]
And a few other pups I can think of from the top of my head

`360.com`: https://mypdns.org/my-privacy-dns/issues/-/issues/1204

`360safe.com.cn`: https://mypdns.org/my-privacy-dns/issues/-/issues/1203

`360totalsecurity.com`: https://mypdns.org/my-privacy-dns/issues/-/issues/63

`getavast.net`: https://mypdns.org/my-privacy-dns/issues/-/issues/62

`avast.com`: https://mypdns.org/my-privacy-dns/issues/-/issues/57

On the other hand these are more Malware and / or Spyware as the lines are getting more blurred over time

**Comment by iam-py-test** [comment]
> And a few other pups I can think of from the top of my head
> 
> `360.com`: https://mypdns.org/my-privacy-dns/issues/-/issues/1204
> 
> `360safe.com.cn`: https://mypdns.org/my-privacy-dns/issues/-/issues/1203
> 
> `360totalsecurity.com`: https://mypdns.org/my-privacy-dns/issues/-/issues/63
> 
> `getavast.net`: https://mypdns.org/my-privacy-dns/issues/-/issues/62
> 
> `avast.com`: https://mypdns.org/my-privacy-dns/issues/-/issues/57
> 
> On the other hand these are more Malware and / or Spyware as the lines are getting more blurred over time

Thanks for the domains.
I will review them in a VM as soon as possible.

**Comment by iam-py-test** [comment]
> 360.com: https://mypdns.org/my-privacy-dns/issues/-/issues/1204

That domain installed so much adware my VM crashed, but considering when it booted back up, it was overrun

**Comment by spirillen** [comment]
> > 360.com: https://mypdns.org/my-privacy-dns/issues/-/issues/1204
> 
> That domain installed so much adware my VM crashed, but considering when it booted back up, it was overrun

:wink:

File: pups.txt (hunk 1 above)
Comment (spirillen) [reply pups.txt:112]: > `https://safeweb.norton.com/reviews?url=winzip.com` 

Is actually spyware....  

```shell
drill -T safeweb.norton.com
net.    172800  IN      NS      m.gtld-servers.net.
trafficmanager.net.     172800  IN      NS      tm1.edgedns-tm.info.
trafficmanager.net.     172800  IN      NS      tm2.edgedns-tm.info.
trafficmanager.net.     172800  IN      NS      tm1.dns-tm.com.
trafficmanager.net.     172800  IN      NS      tm2.dns-tm.com.
shasta-web-prod.trafficmanager.net.     60      IN      A       52.224.14.113
```

`trafficmanager.net`: https://mypdns.org/my-privacy-dns/matrix/-/issues/2507

Comment (spirillen) [reply pups.txt:112]: for the winzip, five it a spin in your virtual install.

Comment (iam-py-test) [reply pups.txt:112]: > > `https://safeweb.norton.com/reviews?url=winzip.com`
> 
> Is actually spyware....
> 
> ```shell
> drill -T safeweb.norton.com
> net.    172800  IN      NS      m.gtld-servers.net.
> trafficmanager.net.     172800  IN      NS      tm1.edgedns-tm.info.
> trafficmanager.net.     172800  IN      NS      tm2.edgedns-tm.info.
> trafficmanager.net.     172800  IN      NS      tm1.dns-tm.com.
> trafficmanager.net.     172800  IN      NS      tm2.dns-tm.com.
> shasta-web-prod.trafficmanager.net.     60      IN      A       52.224.14.113
> ```
> 
> `trafficmanager.net`: https://mypdns.org/my-privacy-dns/matrix/-/issues/2507

@spirillen ?
safeweb.norton.com is owned by Norton - https://www.virustotal.com/gui/url/a72dfee23d9db915aeb3312a0518fcb84fc4eb2c09253145bf9b3a7c4dcadfbc/community

Comment (iam-py-test) [reply pups.txt:112]: @spirillen it is late here so I am logging off my computer and will handle this tomorrow

**Review by fishcharlie — CHANGES_REQUESTED** [review]
Lots of work to be done here. Look at #384 for a good example of how to add a new list.

**Comment by iam-py-test** [comment]
@fishcharlie I am very sorry that I forgot
I have been very busy this week.
Implementing the changes

**Comment by iam-py-test** [comment]
@fishcharlie please re-review

File: pups.txt (hunk 1 above)
Comment (spirillen) [reply pups.txt:112]: norton's safeweb is using the spyware network `trafficmanager.net`, this network is used to pick up any data on you :shrug:

Comment (iam-py-test) [reply pups.txt:112]: @spirillen 
1. That does not mean that their ratings are horrible
2. I have other sources to verify it too
3. Just because a website uses trackers does not mean it is malware. I am pretty sure that would be blocked anyway by uBo

I am very busy this week, so I will not see replies for hours or days

Comment (spirillen) [reply pups.txt:112]: Did blame any of it :smirk: 

I know I wrote `Is actually spyware....` It wasn't ittendent to the safeweb project it self, more it was served over a spooky network.

Sorry if it wasn't clear

**Comment by iam-py-test** [comment]
@fishcharlie is this ready for merge?

**Review by fishcharlie — CHANGES_REQUESTED** [review]
> @fishcharlie @blocklistproject is this ready for merge or do I need to make other changes?

I'm not the one to make the final decision about this. But as it stands right now, I don't support this PR. There is 0 explanation to an average user about what the PUP list should include. What does PUP even stand for?

Additionally, as I mentioned in a previous thread, I do not support relying on 3rd party tools as much for justification for domains being included or not included in these lists.

So no, as it stands I do not support merging this PR.

**Comment by spirillen** [comment]
> Additionally, as I mentioned in a previous thread, I do not support relying on 3rd party tools as much for justification for domains being included or not included in these lists.

Would like to to know if this is the @member opinion as well or it stands by @fishcharlie himself, that sharing information's between project should be a problem rather than a opportunity to improve the each projects by actually share knowledge?

**Comment by fishcharlie** [comment]
@spirillen I have not spoken to @blocklistproject about this. So I can't speak to everyone else's perspective on this. However, there are many considerations here besides just "sharing knowledge". For one, if we are truly relying on those other projects as much as we seem to be, the ethical course of action would be to give them credit in the projects README or something. Not to mention licensing and legal considerations.

"Sharing knowledge" isn't always free. How much have you looked into the licensing requirements for that data?

My point is, it's not as simple as everyone is making it sound like. And I believe we have a responsibility to both follow copyright law, and be ethical in our policies here. However, I haven't spoken to @blocklistproject about this, so I can't comment on official project policy.

**Comment by blocklistproject** [comment]
OK, I have been working a lot lately and have no been as up to date on this pull as I should have I guess.

@spirillen 
1) Where is the PUP list coming from?
1a) Is it a free to use source?

2) What exactly does the PUP list block?

3) What does PUP "stand for"?

4) Do we need an additional list or could these domains be added to a specific list?

@fishcharlie 
1) What are your oppositions to this pull request?

2) What do you believe could be done to fix above stated issues?

3) In your opinion do you believe these domains are safe?
3a) If not, do you believe they could be incorporated into an already existing list(s)?

**Comment by spirillen** [comment]
@fishcharlie mypdns.org is my project, so I know evertything about the license on anything comming from any projects within 

- https://archive.mypdns.org/project/
- https://mypdns.org/explore
- https://github.com/DNS-RPZ
- https://github.com/Import-External-Sources
- https://github.com/My-External-Stuff
- https://github.com/PyFunceble-Templates
- https://github.com/RPZ-DNS
- https://github.com/dns-test
- https://github.com/matrix-rocks
- https://github.com/mypdns
- https://gitlab.com/my-privacy-dns

And they are all covered by this license unless explicit announced https://archive.mypdns.org/w/license/

So how much time did you spend before your comment?

However I do follow you in the comment about licensing, My advice is to do the same as mypdns do only work with other who are fully into open source and open sharing of knowledge.

For the comment of crediting, I don't see what should be changed there. As a open source contributor I will expect no less that this is a obviously thing to do, to credit anyone who have anything to do with your commit, such as a link to the source. such as this example: https://mypdns.org/my-privacy-dns/matrix/-/commit/fa17d0cc54b13c422412c4a877db2cbd4eb91310

image

**Comment by fishcharlie** [comment]
@blocklistproject 

1. I think the questions you asked summarize my opposition to this PR. I think the questions you asked need clear answers, and currently, those answers are not clear.
2. I think someone needs to answer the questions you asked. What does PUP stand for? Where are these initial domains coming from? What is the benefit to having a PUP list over the lists we already have (ie. what benefit does adding this list provide)? If solid answers are given to those questions, I'd be fine with this PR.
3. Unsure. I haven't gone that in-depth yet. That isn't really my issue as it stands now.

**Comment by fishcharlie** [comment]
@spirillen Ok. That answers my question in regards to that. I'm not a lawyer. But I'm pretty sure that license you linked to is NOT compatible with this blocklistproject license. So using mypdns in this project, would violate your license.

I see two paths forward @blocklistproject:

1. @spirillen is able to provide @blocklistproject a custom license to use mypdns in blocklistproject that is compatible with our license.
2. We ban ALL uses of mypdns in regards to this project. Any PR or issue that links to mypdns will be closed and not approved. (This one for example).

**Comment by spirillen** [comment]
> OK, I have been working a lot lately and have no been as up to date on this pull as I should have I guess.
> 
> @spirillen
> 
>     1. Where is the PUP list coming from?
>        1a) Is it a free to use source?
> 
>     2. What exactly does the PUP list block?
> 
>     3. What does PUP "stand for"?
> 
>     4. Do we need an additional list or could these domains be added to a specific list?
> 

Hey @blocklistproject 

1. This PR is owned and started by @iam-py-test 
    > This PR adds the PUP list I requested in #449, and adds some domain from my blocklists to start it off. 
2. See 2) :arrow_up_small: 
3. PUP = Potential Unwanted Programs
4. PUP lists is a "GreyList" it is a lists that of programs that you should be blacklisting, such as Google Chromium as it is SpyWare designed to collect anything about you and send it back to google. https://en.wikipedia.org/wiki/Spyware. In other words a lists of known badware that should be blocked, but for some reason have reached a state where even the old Anti Virus companies have giving in, and unblocked spyware/malware/badware. So something in the middle.

That if you ask for my understanding of the definition

Found a wiki for pup: https://en.wikipedia.org/wiki/Potentially_unwanted_program

**Comment by spirillen** [comment]
> @spirillen Ok. That answers my question in regards to that. I'm not a lawyer. But I'm pretty sure that license you linked to is NOT compatible with this blocklistproject license. So using mypdns in this project, would violate your license.

Are you profiting on this list? if so, then it would be a violation of the general AGPLv3 (modified) license.

And for special license, nope... Please read this little snippet :wink: 

> (1) You are free to use our codes in your production environment as long as you
> don't monetizes from it. If you would like to monetizes from our work, you'll
> need an explicit permission therefor, and it will be based on profit sharing.

The reason is simple, We are all contributing for free in our spare time, so like you, we like that little extra to our coffee and hard drives too.

If you are contributing for free to everyone as your license says: https://github.com/blocklistproject/Lists/blob/master/LICENSE well then there are no issues, but the give back thanks to the source as at least a link or something a-like

Just as you are: https://github.com/blocklistproject/Lists#sponsors

The short version @fishcharlie 

if you economically benefit from this list, you can't get contents from mypdns without sharing the gold. 

If you DO NOT benefit financially from this work, the license says you are all good to receive work from free from mypdns. (Sponsorships to some coffee are all ok)

**Comment by fishcharlie** [comment]
@spirillen Your snippet is only in reference to the "No Commercial use" section. Completely ignoring EVERYTHING else on that page. Same license, state changes, license and copyright notice, disclose source.

All of those things are incompatible with how this project is currently operating and it's license.

Additionally, this projects license does not restrict commercial use like yours does.

As I said, it's incompatible with your license. I standby what I said previously.

**Comment by iam-py-test** [comment]
I am kind-of overwhelmed so please forgive me if I misunderstood anything
I got all but one of these domains from [my blocklists](https://github.com/iam-py-test/my_filters_001) which are CC0.

I have added more of an explanation of what PUPs are in https://github.com/iam-py-test/Lists/commit/b5ea564d2c5915aa9973c292eb6308ff9292315e

**Comment by fishcharlie** [comment]
@blocklistproject I still don't approve of this PR. I personally don't see enough difference between "potentially unwanted programs" and malware to justify having a separate list.

Additionally, just the term "potentially unwanted programs" is extremely broad.

Maybe we can add clarification to these things. But as it stands currently, I'm not sure what is the difference between malware and PUP. And that needs a LOT of clarity before approving this IMO

**Comment by fishcharlie** [comment]
As for the CC0 thing. I think that is perfectly legit, and I approve that. Again not a lawyer, so this is not legal advice. But from my perspective that seems legit and compatible with this projects license, unlike the license that @spirillen linked.

**Comment by iam-py-test** [comment]
@fishcharlie 
Malwarebytes's definition: https://malwarebytes.com/pup
Emsisoft 's definition: https://blog.emsisoft.com/en/2563/what-is-a-pup/
Comodo's definition: https://antivirus.comodo.com/blog/computer-safety/pup-virus/
Also, I am not a lawyer and know very little about law. 
If you feel this PR is useless, I can close it.

**Comment by fishcharlie** [comment]
@iam-py-test So should sourceforge be on this list then?

**Comment by iam-py-test** [comment]
> @iam-py-test So should sourceforge be on this list then?

I think Sourceforge has stopped (https://en.wikipedia.org/wiki/SourceForge#Controversies) and has been removed for uBlock badware.
However, I will willingly add it if it is still hosting PUPs

**Comment by iam-py-test** [comment]
@blocklistproject @fishcharlie I am just going to close this PR and report these domains in an issue
